superset/key_value/utils.py

```diff
@@ -66,10 +68,31 @@ def decode_permalink_id(key: str, salt: str) -> int:
     raise KeyValueParseKeyError(_("Invalid permalink key"))
 
 
-def get_uuid_namespace(seed: str) -> UUID:
-    md5_obj = md5()  # noqa: S324
-    md5_obj.update(seed.encode("utf-8"))
-    return UUID(md5_obj.hexdigest())
+def get_uuid_namespace(seed: str, app: Any = None) -> UUID:
+    """
+    Generate a UUID namespace from a seed string using configured hash algorithm.
+
+    Args:
+        seed: Seed string for namespace generation
+        app: Flask app instance (optional, uses current_app if not provided)
+
+    Returns:
+        UUID namespace
+    """
+    app = app or current_app
+
+    algorithm = app.config["HASH_ALGORITHM"]
+
+    if algorithm == "sha256":
+        sha256_obj = hashlib.sha256()
+        sha256_obj.update(seed.encode("utf-8"))
+        # Use first 16 bytes of SHA-256 digest for UUID
+        return UUID(bytes=sha256_obj.digest()[:16])
+    else:
+        # Legacy MD5 path for backward compatibility
+        md5_obj = md5()  # noqa: S324
+        md5_obj.update(seed.encode("utf-8"))
+        return UUID(md5_obj.hexdigest())
```

Comment:
To keep existing permalinks (or other key-value entries that rely on deterministic UUIDs) from breaking, we will probably need to update all key-value entries from the old format to the new one. See https://github.com/apache/superset/blob/78907d08cd5a3a53e25050a08328bb3b7770f3c2/superset/key_value/utils.py#L56-L66 for the critical code that would break existing permalinks.

I suspect changing this to not break existing permalinks will require a fairly substantial redesign of how permalink ids are encoded/decoded.. 🤔